### PR TITLE
[vcpkg] Restrict telemetry uploads to TLS 1.2

### PR DESCRIPTION
**Pull request by BillyONeal**
Azure Security Policy forbids use of SSL2, SSL3, TLS1.0, or TLS1.1.

### CODE REVIEW

File: toolsrc/CMakeLists.txt [hunk 1]
```diff
@@ -114,6 +114,9 @@ file(GLOB_RECURSE VCPKGLIB_SOURCES CONFIGURE_DEPENDS src/vcpkg/*.cpp)
 add_library(vcpkglib OBJECT ${VCPKGLIB_SOURCES})
 
 add_executable(vcpkg src/vcpkg.cpp $<TARGET_OBJECTS:vcpkglib>)
+if (WIN32)
+    add_executable(vcpkgmetricsuploader WIN32 src/vcpkgmetricsuploader.cpp $<TARGET_OBJECTS:vcpkglib>)
```
Comment (BillyONeal) [inline toolsrc/CMakeLists.txt:118]: @strega-nil @ras0219-msft This is a drive by fix -- the metrics uploader was missing from the cmake version of our builds.

Comment (strega-nil) [inline toolsrc/CMakeLists.txt:118]:
```suggestion
if(WIN32)
```

to be similar to the rest of the file.

File: toolsrc/CMakeLists.txt [hunk 2]
```diff
@@ -114,6 +114,9 @@ file(GLOB_RECURSE VCPKGLIB_SOURCES CONFIGURE_DEPENDS src/vcpkg/*.cpp)
 add_library(vcpkglib OBJECT ${VCPKGLIB_SOURCES})
 
 add_executable(vcpkg src/vcpkg.cpp $<TARGET_OBJECTS:vcpkglib>)
+if (WIN32)
```
Comment (strega-nil) [inline toolsrc/CMakeLists.txt:117]:
```suggestion
if(WIN32)
```

to be similar to the rest of the file

**Review by strega-nil — APPROVED** [review]
LGTM! Please change that one thing

File: toolsrc/CMakeLists.txt (hunk 1 above)
Comment (strega-nil) [reply toolsrc/CMakeLists.txt:118]: do we not have metrics on non-windows users?

Comment (BillyONeal) [reply toolsrc/CMakeLists.txt:118]: @strega-nil On non-Windows we use curl.

Comment (strega-nil) [reply toolsrc/CMakeLists.txt:118]: oh, suddenly things make sense.

Comment (strega-nil) [reply toolsrc/CMakeLists.txt:118]: hmm, we probably want to only build this when `NOT VCPKG_DISABLE_METRICS`; there's no reason to build it if the user isn't sending any metrics.

Comment (BillyONeal) [reply toolsrc/CMakeLists.txt:118]: :sigh: I missed this and merged because it was marked signed off before our meeting, will do a follow up shortly....

**Pull request merged**